hedera-node/src/main/java/com/hedera/services/context/TransactionContext.java

```diff
@@ -232,12 +233,26 @@ default AccountID effectivePayer() {
 
 	/**
 	 * Sets a triggered TxnAccessor for execution
-	 * @param accessor
+	 * @param accessor the accessor which will be triggered
 	 */
 	void trigger(TxnAccessor accessor);
 
 	/**
 	 * Returns a triggered TxnAccessor
 	 */
 	TxnAccessor triggeredTxn();
+
+	/**
+	 * Sets an {@link ExpiringEntity} to be later tracked for purging when expired
+	 * @param expiringEntity the information about entity which will be tracked for future purge
+	 */
+	void setExpiringEntity(ExpiringEntity expiringEntity);
```

Comment:
I'd probably delay adding these methods---that is, `setExpiringEntity()` and `expiringEntity()`---until we have more clarity how how to generalize expiration for all entity types. (Notice for example that a single `ContractCall` transaction can create many accounts.)

Reply:
Since there is a possibility of multiple creations in a single op, switching it to a List<ExpiringEntity> might be fine and this method will append to the list.
Thoughts?

Reply:
Makes sense :)